llama_index/vector_stores/pgvecto_rs.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from typing import Any, List
+
+from llama_index.bridge.pydantic import PrivateAttr
+from llama_index.schema import BaseNode
+from llama_index.vector_stores.types import (
+    BasePydanticVectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from llama_index.vector_stores.utils import metadata_dict_to_node, node_to_metadata_dict
+
+logger = logging.getLogger(__name__)
+import_err_msg = (
+    '`pgvecto_rs.sdk` package not found, please run `pip install "pgvecto_rs[sdk]"`'
+)
+
+
+class PGVectoRsStore(BasePydanticVectorStore):
+    stores_text = True
+
+    _client = PrivateAttr()
+
+    def __init__(self, pgvecto_rs: Any) -> None:
+        try:
+            from pgvecto_rs.sdk import PGVectoRs
+        except ImportError:
+            raise ImportError(import_err_msg)
+        self._client: PGVectoRs = pgvecto_rs
+        super().__init__()
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "PGVectoRsStore"
+
+    @property
+    def client(self) -> Any:
+        return self._client
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+    ) -> List[str]:
+        from pgvecto_rs.sdk import Record
+
+        records = [
+            Record(
+                id=node.id_,
+                text="",
```

Comment:
is the text here empty?

Reply:
Yes, the text field is not needed for  llamaindex